core/src/main/java/jenkins/mvn/GlobalMavenConfig.java

```diff
@@ -17,6 +19,11 @@ public GlobalMavenConfig() {
     }
 
     @Override
+    public ToolConfigurationCategory getCategory() {
+        return GlobalConfigurationCategory.get(ToolConfigurationCategory.class);
```

Comment:
So I think this is the potential argument against `ToolConfigurationCategory` being in `jenkins.tools`—that it might be used by things which are not actually `ToolDescriptor`s but something conceptually related. On the other hand I would expect that any plugin which is defining such configuration would also be defining `ToolDescriptor`s and thus implicitly using `ToolConfigurationCategory`.